src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -300,33 +305,87 @@ private static void SetProviderConfiguration(string strConfig, EventPipeConfigur
                 throw new ArgumentNullException(nameof(strConfig));
             }
 
-            // String must be of the form "providerName:keywords:level,providerName:keywords:level..."
-            string[] providers = strConfig.Split(ProviderConfigDelimiter);
+            // String must be of the form "providerName:keywords:level:parameter,providerName:keywords:level:parameter..."
+            // where parameter is of the form: "key=value;key=value..."
+            string[] providers = strConfig.Split(
+                ProviderConfigDelimiter,
+                StringSplitOptions.RemoveEmptyEntries); // Remove "empty" providers.
             foreach (string provider in providers)
             {
-                string[] components = provider.Split(ConfigComponentDelimiter);
-                if (components.Length == 3)
-                {
-                    string providerName = components[0];
+                const char FilterDataDelimiter = ';';
+                string[] components = provider.Split(
+                    ConfigComponentDelimiter,
+                    StringSplitOptions.None); // Keep empty tokens
 
-                    // We use a try/catch block here because ulong.TryParse won't accept 0x at the beginning
-                    // of a hex string.  Thus, we either need to conditionally strip it or handle the exception.
-                    // Given that this is not a perf-critical path, catching the exception is the simpler code.
-                    ulong keywords = 0;
-                    try
-                    {
-                        keywords = Convert.ToUInt64(components[1], 16);
-                    }
-                    catch { }
+                string providerName = null;
+                ulong keywords = 0;
+                uint level = 0;
```

Comment:
I realize that you are keeping existing behavior, but I generally it is good to have defaults where keywords are ulong.Maxvalue and level = 5 (verbose).  Otherwise the shortcut of omitting them is pretty pointless (because it turns everything off).